### DISCUSSION TITLE
Format the text of helpers.relativeTime

### DISCUSSION THREAD

**Discussion by DanDixon** [custom-cards/button-card#753] [Q&A]
I'd like to shorten the relativeTime output so it fits in a smaller space:

![image](https://github.com/custom-cards/button-card/assets/10511776/3cb23ce0-a8cb-4d59-a764-1647f1d01293)

left = relativeTime 
right = custom code that I was hoping relativeTime would replace

I'd like to do something like this:
```
        var rt = helpers.relativeTime(states['input_datetime.porch_arrived'].state)
        rt = rt.replace("hours","h")
        return str;
```
But relativeTime:
> Returns an lit-html template which will render a relative time and update automatically.

And so you can't make any change to the text output of relativeTime apparently.

For example:
```
        var rt = helpers.relativeTime(states['input_datetime.porch_arrived'].state)
        return "x" + rt;
```
Shows this:
![image](https://github.com/custom-cards/button-card/assets/10511776/c7965703-3e0c-4cdf-936d-759cc49b41ba)

Any workarounds for this?

**Comment by pavlovs-hub** [comment]
Hi DanDixon, did you find a solution to your issue?

**Comment by roblomq** [comment]
Have the same sort question.
Where

```
    label: >
     [[[
      var TimeAgo = helpers.relativeTime(states['binary_sensor.garagedeur_any'].last_changed);
      return (TimeAgo)
     ]]]
```
Gives
![2024-08-09_20h13_01](https://github.com/user-attachments/assets/e4d6c3fa-64a9-4f85-9b3e-33da25e3ffee)

While
```
    label: >
     [[[
      var TimeAgo = helpers.relativeTime(states['binary_sensor.garagedeur_any'].last_changed);
      return 'Door closed ' + (TimeAgo) + ' ago.'
     ]]]
```

gives
![2024-08-09_20h11_26](https://github.com/user-attachments/assets/8fbc4ea6-294c-4869-bf4c-ef959ea162df)

**Reply by ngocjohn** [reply]
https://github.com/custom-cards/button-card/discussions/753#discussioncomment-10292751

**Comment by ngocjohn** [comment]
But last_changed is for every state change, in your case also in case the door has opened...
use this function to convert datetime object to string.. 
```yaml
      - type: custom:button-card
        entity: light.ceiling_lights
        name: Test Light
        show_state: true
        show_label: true
        label: >
          [[[
            if (entity !== undefined) {
                let timeAgo = states['light.ceiling_lights'].last_changed;
                //console.log(timeAgo);

                function timeSince(date) {
                    const now = new Date();
                    const secondsPast = (now - new Date(date)) / 1000;

                    if (secondsPast < 60) {
                        return `${Math.floor(secondsPast)} seconds`;
                    }
                    if (secondsPast < 3600) {
                        return `${Math.floor(secondsPast / 60)} minutes`;
                    }
                    if (secondsPast < 86400) {
                        return `${Math.floor(secondsPast / 3600)} hours`;
                    }
                    if (secondsPast < 2592000) {
                        return `${Math.floor(secondsPast / 86400)} days`;
                    }
                    if (secondsPast < 31536000) {
                        return `${Math.floor(secondsPast / 2592000)} months`;
                    }
                    return `${Math.floor(secondsPast / 31536000)} years`;
                }
                var timeAgoStr = timeSince(timeAgo);
                return 'Light state changed ' + timeAgoStr + ' ago';
            }
          ]]]

```

**Comment by Mariusthvdb** [comment]
But did you check the format helpers? There a quite a few of them and no need for the super complex templates at all.

Most is simply built right in
Check just above the Styles paragraph 
https://github.com/custom-cards/button-card?tab=readme-ov-file#styles
